src/renderer/components/+network-ingresses/ingress-details.tsx

```diff
@@ -65,44 +65,51 @@ export class IngressDetails extends React.Component<Props> {
   }
 
   renderPaths(ingress: Ingress) {
-    const { spec: { rules }} = ingress;
-
-    if (!rules || !rules.length) return null;
-
-    return rules.map((rule, index) => {
-      return (
-        <div className="rules" key={index}>
-          {rule.host && (
-            <div className="host-title">
-              <>Host: {rule.host}</>
-            </div>
-          )}
-          {rule.http && (
-            <Table className="paths">
-              <TableHead>
-                <TableCell className="path">Path</TableCell>
-                <TableCell className="backends">Backends</TableCell>
-              </TableHead>
-              {
-                rule.http.paths.map((path, index) => {
-                  const { serviceName, servicePort } = getBackendServiceNamePort(path.backend);
-                  const backend = `${serviceName}:${servicePort}`;
+    const { spec: { rules = [], tls = [] }} = ingress;
+    const protocol = tls.length === 0
+      ? "http"
+      : "https";
+
+    return rules.map((rule, index) => (
+      <div className="rules" key={index}>
+        {rule.host && (
+          <div className="host-title">
+            <>Host: {rule.host}</>
+          </div>
+        )}
+        {rule.http && (
+          <Table className="paths">
+            <TableHead>
+              <TableCell className="path">Path</TableCell>
+              <TableCell className="link">Link</TableCell>
+              <TableCell className="backends">Backends</TableCell>
+            </TableHead>
+            {
+              rule.http.paths
+                .map(({ path = "/", backend }, index) => {
+                  const link = `${protocol}://${rule.host || "*"}${path}`;
```

Comment:
if host is `*`, isn't hyperlink broken then?

Reply:
Yes it is, I guess I should then not display them as links then.